plugin/main/src/kotlinx/benchmark/gradle/internal/BenchmarkDependencies.kt

```diff
@@ -0,0 +1,79 @@
+package kotlinx.benchmark.gradle.internal
+
+import kotlinx.benchmark.gradle.BenchmarksExtension
+import org.gradle.api.*
+import org.gradle.api.artifacts.*
+import org.gradle.api.attributes.*
+import org.gradle.api.attributes.Usage.*
+import org.gradle.api.model.*
+import org.gradle.util.GradleVersion
+
+/**
+ * Utility for containing all Gradle [Configuration]s used by Benchmarks.
+ */
+internal class BenchmarkDependencies(
+    project: Project,
+    benchmarksExtension: BenchmarksExtension,
+) {
+    private val objects: ObjectFactory = project.objects
+
+    private val benchmarkGenerator: Configuration =
+        project.configurations.create("benchmarkGenerator") {
+            it.description =
+                "Internal Kotlinx Benchmarks Configuration. Contains declared dependencies required for running benchmark generators."
+            it.declarable()
+
+            it.defaultDependencies { deps ->
+                deps.addLater(
+                    benchmarksExtension.kotlinCompilerVersion.map { version ->
+                        project.dependencies.create("org.jetbrains.kotlin:kotlin-compiler-embeddable:$version")
+                    }
+                )
+            }
+        }
+
+    val benchmarkGeneratorResolver: Configuration =
+        project.configurations.create("benchmarkGenerator.resolver") {
+            // The name has a dot, to prevent Gradle from generating a Kotlin DSL accessor.
+            it.description =
+                "Internal Kotlinx Benchmarks Configuration. Resolves dependencies required for running benchmark generators."
+            it.resolvable()
+
+            it.extendsFrom(benchmarkGenerator)
+
+            it.attributes { atts ->
+                atts.attribute(USAGE_ATTRIBUTE, objects.named(Usage::class.java, JAVA_RUNTIME))
+            }
+        }
+
+    internal companion object {
+
+        /** Mark this [Configuration] as one that will be used to declare dependencies. */
+        private fun Configuration.declarable() {
+            isCanBeDeclaredCompat = true
+            isCanBeResolved = false
+            isCanBeConsumed = false
+            isVisible = false
```

Comment:
What's the reason for hiding these configurations? Wouldn't it be helpful for build scripts debugging to keep these configs visible?

Reply:
`isVisible` is confusing, its Javadoc doesn't make sense, it's misnamed, and its default should really be `false`.

If a Configuration is 'visible', then it could automatically be added to the subproject's default outgoing archive, and the producing tasks will always be executed. It should always be set to `false`, unless there's an extremely good reason.

https://github.com/gradle/gradle/issues/6875#issuecomment-1084140718

A better name would probably be `isPrimaryOutput()` and the Javadoc would say "Only enable `isPrimaryOutput()` on outgoing Configurations. The contents will always be shared to consumers by default, unless performing variant aware matching, and they request a specific, valid, and available variant."

Reply:
Thank you for the clarification!